src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -416,7 +407,7 @@ static uint32_t get_tunnel_image(ObjectEntryIndex index, uint8_t type)
     if (obj != nullptr)
     {
         auto tobj = static_cast<TerrainEdgeObject*>(obj);
-        hasDoors = tobj->HasDoors;
+        hasDoors = tobj->HasDoors && !tobj->UsesFallbackImages();
```

Comment:
why this change?

Reply:
Because the fallback sprites lack doors. If I don’t include this, it will glitch heavily.

Reply:
Should this become a flag of the object then?

Reply:
It is - `UsesFallbackImages()` is simply a getter for a flag. Or is that not what you meant?